code/game/objects/effects/temporary_visuals/miscellaneous.dm

```diff
@@ -6,16 +6,22 @@
 /obj/effect/temp_visual/explosion
 	name = "explosion"
 	icon = 'icons/effects/explosion.dmi'
-	icon_state = "explosion"
+	icon_state = ""
 	duration = 8
-	pixel_x = -16
-	pixel_y = -16
 	light_system = STATIC_LIGHT
+	///The icon for the explosion itself
+	var/explosion_icon = "explosion"
 
 /obj/effect/temp_visual/explosion/Initialize(mapload, radius, color)
 	. = ..()
 	set_light(radius, radius, color)
 
+	var/image/I = image(icon, src, explosion_icon, 10, pixel_x = -16, pixel_y = -16)
+	var/matrix/rotate = matrix()
+	rotate.Turn(rand(0, 359))
+	I.transform = rotate
+	add_overlay(I) //we use an overlay so the explosion and light source are both in the correct location
```

Comment:
can we just overlays += I here to microopt

Reply:
done